### ISSUE TITLE
chroot: failed to run command 'dpkg': No such file or directory

### ISSUE THREAD

**Issue by sh1sh1n11** [Volt-cx/arch-nano#1]
Hi Yarpii,

Thank you so very much for writing this guide!

Other than the [Jetson/Porting Arch guide](https://elinux.org/Jetson/Porting_Arch), yours is the only one that I could find on the internet.

I followed your instructions to the word, except for I used updated copies of both the Arch Linux Arm kernel (downloaded as of 2022-01-10 at 14:35 hrs UTC ) and the L4T drivers package (Jetson-210_Linux_R32.6.1_aarch64.tbz2).

When I run the "sudo ./apply_binaries.sh" command to run the script after all the prep work has been done, I receive the following error message.

chroot: failed to run command 'dpkg': No such file or directory

Below is the terminal output.

`sudo ./apply_binaries.sh 
Using rootfs directory of: /home/shashank/jetson/Linux_for_Tegra/rootfs
Installing extlinux.conf into /boot/extlinux in target rootfs
/home/shashank/jetson/Linux_for_Tegra/nv_tegra/nv-apply-debs.sh
Root file system directory is /home/shashank/jetson/Linux_for_Tegra/rootfs
Copying public debian packages to rootfs
Start L4T BSP package installation
QEMU binary is not available, looking for QEMU from host system
Found /usr/bin/qemu-aarch64-static
Installing QEMU binary in rootfs
~/jetson/Linux_for_Tegra/rootfs ~/jetson/Linux_for_Tegra
Installing BSP Debian packages in /home/shashank/jetson/Linux_for_Tegra/rootfs
chroot: failed to run command 'dpkg': No such file or directory`

Any help / pointers would be greatly appreciated! And thank you so much once again :)

**Comment by sh1sh1n11** [comment]
So I tried the entire process once again with L4T 32.5.1 and received the same error "chroot: failed to run command 'dpkg': No such file or directory" when I executed the script "apply_binaries.sh".

**Comment by sh1sh1n11** [comment]
- So I did a little bit of digging through the "apply_binaries.sh" shell script, which led me to the "Linux_for_Tegra/nv_tegra/nv-apply-debs.sh" shell script.

This is the part inside the "nv-apply-debs.sh" shell script that seems to be the problem.

`echo "Installing BSP Debian packages in ${L4T_ROOTFS_DIR}"
if [ "${#pre_deb_list[@]}" -ne 0 ]; then
        LC_ALL=C PYTHONHASHSEED=0 chroot . dpkg -i --path-include="/usr/share/doc/*" "${pre_deb_list[@]}"
fi
LC_ALL=C PYTHONHASHSEED=0 chroot . dpkg -i --path-include="/usr/share/doc/*" "${deb_list[@]}"
rm -f "${L4T_ROOTFS_DEB_DIR}/.nv-l4t-disable-boot-fw-update-in-preinstall"
popd`

**Comment by sh1sh1n11** [comment]
Tried it on a Debian 11 machine also. Same error "chroot: failed to run command 'dpkg': No such file or directory".

**Comment by kerojiang** [comment]
hi  
Have you sloved this problem?
I have meet the same problem.

**Comment by theSkele** [comment]
I had same issue, Host is arch. I resolved by simply installing dpkg with pacman
Idk why I thought it was something other than not having it installed but it took me a day to realize.

No idea if this applies to your host running Ubuntu Server though (I would assume it would already have dpkg)

**Comment by Yarpii** [comment]
Hi there, sorry for really really late delay with answer... I will start this project again on my jetson and try to update this one again... But sadly it was a shitty years 2022. 
Also would i like to notice that projects of Volt.cx would be moved to a other Github Org..
Again im so so so sorry

**Comment by theSkele** [comment]
> Hi there, sorry for really really late delay with answer... I will start this project again on my jetson and try to update this one again... But sadly it was a shitty years 2022. Also would i like to notice that projects of Volt.cx would be moved to a other Github Org.. Again im so so so sorry

I have forked the repo and updated info accordingly. Updated links, fixed typos, added images, along with steps on how to properly  pre-configure users/groups within RootFS

**Comment by theSkele** [comment]
> Hi @Yarpii
> 
> Thank you so very much for writing this guide!
> 
> Other than the [Jetson/Porting Arch guide](https://elinux.org/Jetson/Porting_Arch), yours is the only one that I could find on the internet.
> 
> I followed your instructions to the word, except for I used updated copies of both the Arch Linux Arm kernel (downloaded as of 2022-01-10 at 14:35 hrs UTC ) and the L4T drivers package (Jetson-210_Linux_R32.6.1_aarch64.tbz2).
> 
> When I run the "sudo ./apply_binaries.sh" command to run the script after all the prep work has been done, I receive the following error message.
> 
> chroot: failed to run command 'dpkg': No such file or directory
> 
> Below is the terminal output.
> 
> `sudo ./apply_binaries.sh Using rootfs directory of: /home/ubuntu/jetson/Linux_for_Tegra/rootfs Installing extlinux.conf into /boot/extlinux in target rootfs /home/ubuntu/jetson/Linux_for_Tegra/nv_tegra/nv-apply-debs.sh Root file system directory is /home/ubuntu/jetson/Linux_for_Tegra/rootfs Copying public debian packages to rootfs Start L4T BSP package installation QEMU binary is not available, looking for QEMU from host system Found /usr/bin/qemu-aarch64-static Installing QEMU binary in rootfs ~/jetson/Linux_for_Tegra/rootfs ~/jetson/Linux_for_Tegra Installing BSP Debian packages in /home/ubuntu/jetson/Linux_for_Tegra/rootfs chroot: failed to run command 'dpkg': No such file or directory`
> 
> Any help / pointers would be greatly appreciated! And thank you so much once again :)
> 
> PS - I am trying to compile this on Ubuntu 18.04.6 LTS server edition. Sadly, I received the same error on L4T R32.6.1 also.

Im Sorry my reply of "installing dpkg via pacman" was completely wrong.

DPKG must be installed in the rootfs (which the Sample RootFS contains, that nvidia provides) but since we dont use the SampleRootFS its not there.

Older apply_binaries.sh scripts apparently never used dpkg and simply used tarballs.
BUT luckily the apply_binaries.sh script takes the flag --target-overlay which reverts back to tarballs and DPKG is not neccessary.

TLDR: `sudo ./apply_binaries.sh --target-overlay`
fixes your problem
